mesonbuild/dependencies/dub.py

```diff
@@ -59,158 +58,318 @@ def __init__(self, name: str, environment: 'Environment', kwargs: T.Dict[str, T.
         mlog.debug('Determining dependency {!r} with DUB executable '
                    '{!r}'.format(name, self.dubbin.get_path()))
 
+        # if an explicit version spec was stated, use this when querying Dub
+        main_pack_spec = name
+        if 'version' in kwargs:
+            main_pack_spec = f'{name}@{kwargs["version"]}'
```

Comment:
For the `version` kwarg, currently we can only specify an exact version like `0.9.4`, otherwise Meson complains about version mismatch.
However it would be possible to use whatever DUB normally accepts such as `~>0.9.3`.

Reply:
I think I'm reading the documentation right that `~>0.9.3` would be equivalent to the meson version expression `['>= 0.9.3', '< 0.10']`?